src/core/job.c

```diff
@@ -1355,10 +1355,12 @@ void job_shutdown_magic(Job *j) {
 
 int job_get_timeout(Job *j, usec_t *timeout) {
         usec_t x = USEC_INFINITY, y = USEC_INFINITY;
-        Unit *u = j->unit;
+        Unit *u;
         int r;
 
-        assert(u);
+        assert(j);
+
+        u = ASSERT_PTR(j->unit);
```

Comment:
```c
u = ASSERT_PTR(ASSERT_PTR(j)->unit);`
```

?

Reply:
I considered this but I decided I don't like it. It just seems to opaque to me... (Whether it even works is another question.)